src/NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1797,7 +1797,7 @@ private static int CompareBigEndian(byte[] c1, int ic1, int c1len, byte[] c2, in
         }
 
 
-        public static bool IsValidSignatureEncoding(byte[] sig)
+        public static bool IsValidSignatureEncoding(byte[] sig, bool haveSigHash)
```

Comment:
I think this needs to be by default true so `bool haveSigHash = true` and only from the method `IsCanonicalBlockSignature` we call it with false (thats when we validate POS blocks)